block/serialize_test.go

```diff
@@ -0,0 +1,78 @@
+package block
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"testing"
+
+	"github.com/vulpemventures/go-elements/internal/bufferutil"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBlockSerialization(t *testing.T) {
+	file, err := ioutil.ReadFile("testdata/deserialize.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	var tests []map[string]interface{}
+	err = json.Unmarshal(file, &tests)
+
+	for _, v := range tests {
+		testName := v["name"].(string)
+		t.Run(testName, func(t *testing.T) {
+			block, err := NewFromHex(v["hex"].(string))
+			if err != nil {
+				t.Errorf("test: %v, err: %v", testName, err)
+			}
+
+			serializeBlock, err := block.SerializeBlock()
+			if err != nil {
+				t.Error(err)
+			}
+
+			assert.Equal(t, v["hex"].(string), hex.EncodeToString(serializeBlock))
+		})
+	}
+}
+
+func TestBlockHeaderSerialization(t *testing.T) {
```

Comment:
The previous test should already test the header serialization, shouldn't it?
Is this test necessary?

Reply:
You are right, this test case was helping me to identify mistakes easier since transaction was not included.
I will remove it.

Reply:
Done